core/field_image.js

```diff
@@ -40,60 +40,60 @@ goog.require('Blockly.utils.Size');
  * @param {!(string|number)} width Width of the image.
  * @param {!(string|number)} height Height of the image.
  * @param {string=} opt_alt Optional alt text for when block is collapsed.
- * @param {Function=} opt_onClick Optional function to be called when the image
- *     is clicked.  If opt_onClick is defined, opt_alt must also be defined.
+ * @param {function(!Blockly.FieldImage)=} opt_onClick Optional function to be
```

Comment:
Nice change, I'd keep Blockly.FieldImage as optional. ie: they don't have to define a function that takes in a FieldImage if they didn't want to. A function that we call when the the image is clicked would suffice: 
```
@param {function(Blockly.FieldImage)=}
```

Reply:
Should it be `{function(Blockly.FieldImage)=}` or `{function(!Blockly.FieldImage=)=}`? The former is definitely more readable hehe.

Reply:
Yep, you're right! ``{function(!Blockly.FieldImage=)=}`` is the right way to describe the above.
On second thoughts, in the interest of simplicity, maybe just keep it as ``{function(!Blockly.FieldImage)=}``.